### PR TITLE
Support job level annotations; fix jobs scheduling config

### PR DESCRIPTION
**Pull request by jedcunningham**
Add job level annotations as some tooling needs to be able to add them.

Also fix jobs to use their own nodeSelector, affinity, and
tolerations.

Related to #16291

### CODE REVIEW

**Comment by jedcunningham** [comment]
cc @dnskr

File: chart/templates/jobs/create-user-job.yaml [hunk 1]
```diff
@@ -19,9 +19,9 @@
 ## Airflow Create User Job
 #################################
 {{- if .Values.webserver.defaultUser.enabled }}
-{{- $nodeSelector := or .Values.webserver.nodeSelector .Values.nodeSelector }}
-{{- $affinity := or .Values.webserver.affinity .Values.affinity }}
-{{- $tolerations := or .Values.webserver.tolerations .Values.tolerations }}
+{{- $nodeSelector := or .Values.createUserJob.nodeSelector .Values.nodeSelector }}
```
Comment (kaxil) [inline chart/templates/jobs/create-user-job.yaml:22]: Worth adding in Updating.md?

**Comment by dnskr** [comment]
It works! 💯 

I have tested your commit with Airflow application manifest
```yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: airflow
  finalizers:
    - resources-finalizer.argocd.argoproj.io
spec:
  project: default
  source:
    repoURL: https://github.com/astronomer/airflow.git
    targetRevision: fix_jobs_spec
    path: chart
    helm:
      version: v3
      valueFiles:
      - values.yaml
      values: |
        executor: KubernetesExecutor
        postgresql:
          enabled: true
          persistence:
            enabled: false
        flower:
          enabled: false
        redis:
          enabled: false
        statsd:
          enabled: false
        pgbouncer:
          enabled: false
        migrateDatabaseJob:
          jobAnnotations:
            "argocd.argoproj.io/hook": Sync
            "argocd.argoproj.io/sync-wave": "0"
            "argocd.argoproj.io/hook-delete-policy": BeforeHookCreation,HookSucceeded
  destination:
    server: https://kubernetes.default.svc
    namespace: default
  syncPolicy:
    automated:
      prune: true
      selfHeal: true

```

File: chart/templates/jobs/create-user-job.yaml (hunk 1 above)
Comment (jedcunningham) [reply chart/templates/jobs/create-user-job.yaml:22]: Might as well, incoming.

**Review by kaxil — APPROVED** [review]

**Pull request merged**